1070062 - Traceback when getting instances of LMI_MDRAIDFormat of deactivated RAID on multipath devices

commit 24e96f49be819e34f06bcaca6c9546de7ed6a5fa
Author: Jan Safranek <jsafrane@redhat.com>
Date:   Wed Feb 26 10:09:46 2014 +0100

    Don't use MDRaidMember.uuid, it's wrong.
    
    See rhbz#1070095, blivet's MDRaidMember.uuid reports UUID of the MD RAID, not
    UUID of the member. FormatProvider.get_format_id() would use it, so we need
    to override it in its LMI_MDRAIDFormatProvider subclass.

diff --git a/src/lmi/storage/LMI_MDRAIDFormatProvider.py b/src/lmi/storage/LMI_MDRAIDFormatProvider.py
index 99d6749..df7b8f8 100644
--- a/src/lmi/storage/LMI_MDRAIDFormatProvider.py
+++ b/src/lmi/storage/LMI_MDRAIDFormatProvider.py
@@ -30,6 +30,7 @@ LMI_MDRAIDFormatProvider
 from lmi.storage.FormatProvider import FormatProvider
 import blivet.formats.mdraid
 import lmi.providers.cmpi_logging as cmpi_logging
+from lmi.storage.util import storage
 
 class LMI_MDRAIDFormatProvider(FormatProvider):
     """
@@ -49,6 +50,18 @@ class LMI_MDRAIDFormatProvider(FormatProvider):
         return False
 
     @cmpi_logging.trace_method
+    def get_format_id(self, device, fmt):
+        """
+            Return LMI_DataFormat.Name. The name should be unique and stable
+            across reboots or reconfigurations.
+
+            This is workaround for MDRaidMember reporting wrong UUID, see
+            https://bugzilla.redhat.com/show_bug.cgi?id=1070095
+        """
+        # TODO: remove this method when 1070095 is fixed
+        return "DEVICE=" + storage.get_persistent_name(device)
+
+    @cmpi_logging.trace_method
     def get_instance(self, env, model, fmt=None):
         """
             Get instance.
